### PR TITLE
change access to location database

### PR DESCRIPTION
**Pull request by Mdominykas**
● Moved entire Location database logic to LocationDatabaseAccess
● Added statement that removes previous database to App.xml.cs (after everyone changes databases it should be removed)
● Moved database names to DatabaseAccessConstants.cs class
● Renamed DataStore functions to fit for other databases (not implemented for others)
● Renamed from MockDataStore to LocationDataStore, as word Mock is ussually used for classes in unit testing

It also has a very weird bug, which I could not fix:
If new Locations are added from code, then they appear correctly on map (thus data is correct in the database), however when Location details, it shows incorrect coordinates. It only last until you add a new location. After drop tables command will be removed from App.xml.cs, it will also not happen after app has been restarted

### CODE REVIEW

**Review by MigleMic — APPROVED** [review]
Works great
I also think we should add more info about locations. If it's a building, a monument and etc.

File: Encountify/Services/IDataStore.cs [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
 {
+    //generic class for every needed place
     public interface IDataStore<T>
     {
-        Task<bool> AddLocationAsync(T location);
-        Task<bool> UpdateLocationAsync(T location);
-        Task<bool> DeleteLocationAsync(int id);
-        Task<T> GetLocationAsync(int id);
-        Task<IEnumerable<T>> GetLocationsAsync(bool forceRefresh = false);
+        Task<bool> AddAsync(T location);
```
Comment (hickoryeyes) [inline Encountify/Services/IDataStore.cs:9]: If IDataStore is now universal, shouldn't the variables also be renamed? Like T location to T item

File: Encountify/Services/LocationDatabaseAccess.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    //it may can be made generic for all databases and or merged with IDataStore
+    //however it might become a huge hussle if we when adding databases with different logic
+    class LocationDatabaseAccess
+    {
+        public List<Location> GetLocationList()
+        {
+            string dbPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Personal), 
+                DatabaseAccessConstants.LocationDatabaseName);
+            SQLiteConnection db = new SQLiteConnection(dbPath);
+            //haven't tested it, however we may not need to create new SQLiteConnection in every request
```
Comment (hickoryeyes) [inline Encountify/Services/LocationDatabaseAccess.cs:19]: You're right. It's not necessary if we don't use db.close();

**Review by hickoryeyes — CHANGES_REQUESTED** [review]

**Review by Waltzerizm — APPROVED** [review]

File: Encountify/Services/IDataStore.cs (hunk 1 above)
Comment (Mdominykas) [reply Encountify/Services/IDataStore.cs:9]: Thanks, totally missed it

File: Encountify/Services/LocationDatabaseAccess.cs (hunk 2 above)
Comment (Mdominykas) [reply Encountify/Services/LocationDatabaseAccess.cs:19]: Could you link the source that says that it is the only time that new connection must be generated. As far I have tested it does work flawlessly, though if I remove it and some operations does invalidate SQLiteConnection, then we might run into some very strange and hard to debug issues later on.

**Review by hickoryeyes — APPROVED** [review]

File: Encountify/Services/LocationDatabaseAccess.cs (hunk 2 above)
Comment (hickoryeyes) [reply Encountify/Services/LocationDatabaseAccess.cs:19]: image

**Pull request merged**